frontends/PyCDE/src/pycde/system.py

```diff
@@ -35,26 +37,36 @@ class System:
   output SystemVerilog."""
 
   __slots__ = [
-      "mod", "modules", "passed", "_module_symbols", "_symbol_modules",
-      "_old_system_token", "_symbols", "_generate_queue", "_primdb"
+      "mod", "modules", "name", "passed", "_module_symbols", "_symbol_modules",
+      "_old_system_token", "_symbols", "_generate_queue", "_primdb",
+      "_output_directory"
   ]
 
-  passes = [
-      "lower-msft-to-hw", "lower-seq-to-sv", "hw.module(prettify-verilog)",
-      "hw.module(hw-cleanup)"
-  ]
+  PASSES = """
+    lower-msft-to-hw{{tops={tops} verilog-file={verilog_file} tcl-file={tcl_file}}},
+    lower-seq-to-sv,hw.module(prettify-verilog),hw.module(hw-cleanup)
+  """
 
-  def __init__(self, modules, primdb: PrimitiveDB = None):
+  def __init__(self,
+               modules,
+               primdb: PrimitiveDB = None,
+               name: str = "PyCDESystem",
+               output_directory: str = None):
```

Comment:
@teqdruid ended up adding this output_directory for the tests. For the tests to be stable, we want to always write the output to the `%t` directory lit provides, and ensure `%t` is cleaned up of previous runs before running the tests. Since `%t` isn't just os.getcwd() when the tests run, we need to pass it explicitly. It still defaults to os.getcwd() for normal use.